caffe2/opt/optimize_ideep.cc

```diff
@@ -459,15 +467,130 @@ void setPoolingInferenceMode(repr::NNModule *nn) {
   }
 }
 
-void OptimizeForIdeep(
-    repr::NNModule* nn,
-    caffe2::Workspace* ws,
-    bool training_mode) {
-  if (training_mode) {
-    // Only support inference so far
-    return;
+// Pre-convert filters format to expected one here
+// in order to avoid boring conversions during computations
+void preConvertFiltersFormat(repr::NNModule *nn, caffe2::Workspace *ws) {
+  for (auto &node : nn->dataFlow.getMutableNodes()) {
+    if (!repr::nn::is<repr::ConvTranspose>(node) &&
+        !repr::nn::is<repr::Conv>(node) && !repr::nn::is<repr::FC>(node)) {
+      continue;
+    }
+
+    auto *nnOp = repr::nn::get<repr::NeuralNetOperator>(node);
+    if (!isOnIdeepDevice(*nnOp)) {
+      LOG(WARNING) << "Not a IDEEP operator";
+      continue;
+    }
+
+    auto inputs = repr::nn::getInputs(node);
+    if (inputs.size() < 2) {
+      LOG(WARNING) << "Invalid input size";
+      continue;
+    }
+
+    auto *filterBlob = getBlob(inputs[1], ws);
+    auto *filter = getMutableTensor<itensor>(filterBlob);
+    if (filter == nullptr) {
+      CAFFE_ENFORCE(filter, "Filter not a IDEEP tensor");
```

Comment:
We don't need to throw an exception here. Log info is enough. Because it valid user input.